java/com/google/openlocationcode/OpenLocationCode.java

```diff
@@ -496,7 +498,7 @@ public static boolean isValidCode(String code) {
     if (code == null || code.length() < 2) {
       return false;
     }
-    code = code.toUpperCase();
+    code = code.toUpperCase(Locale.ROOT);
```

Comment:
If you wanted to remove _all_ redundant calls of `toUpperCase()`, it could be modified to something like this
```
public static boolean isValidCode(String code) {
    return isValidCodeUpperCase(code.toUpperCase(Locale.ROOT));
}

private static boolean isValidCodeUpperCase(String code) {
    // do validity checks without calling code.toUpperCase() redundantly
}
```
And the constructor would instead call `!isValidCodeUpperCase(newCode)`.
Just a minor improvement...

Reply:
Thanks for pointing out the redundant call. It's so redundant, in fact, that I think we can just remove the one in the constructor since the code will be cast to uppercase in isValidCode().